File: A_Sleuth.py
s = input().replace(' ', '')
if s[len(s) - 2] == 'a' or s[len(s) - 2] == 'e' or s[len(s) - 2] == 'o' or s[len(s) - 2] == 'u' or s[len(s) - 2] == 'y' or s[len(s) - 2] == 'A' or s[len(s) - 2] == 'E' or s[len(s) - 2] == 'O' or s[len(s) - 2] == 'U' or s[len(s) - 2] == 'Y':
    print('YES')
else:
    print('NO')

File: A_Sleuth.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    string s; cin >> s;
    s.erase(remove(s.begin(), s.end(), ' '), s.end());
    cout << s << " ";
    if(s[s.length() - 2] == 'a' || s[s.length() - 2] == 'e' || s[s.length() - 2] == 'o' || s[s.length() - 2] == 'u' || s[s.length() - 2] == 'y' || s[s.length() - 2] == 'A' || s[s.length() - 2] == 'E' || s[s.length() - 2] == 'O' || s[s.length() - 2] == 'U' || s[s.length() - 2] == 'Y'){
        cout << "YES" << "\n";
    }
    else {
        cout << "NO" << "\n";
    }
    return 0;
}